infrastructures/kubernetes/pom.xml

```diff
@@ -114,6 +114,10 @@
             <groupId>org.eclipse.che.core</groupId>
             <artifactId>che-core-db</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.che.infrastructure.docker</groupId>
+            <artifactId>docker-client</artifactId>
```

Comment:
I think this dependency can potentially bring some issues in future. I suggest to move docker auth related code in separate maven module and replace this dependency with it.
@sleshchenko @garagatyi @gazarenkov wdyt?

Reply:
@skabashnyuk the same applies to the `docker-environment` dependency that is just below right?

Reply:
I would say that not the same. It is supposed that `docker-environment` may be reused by another infrastructure.

Reply:
Yes. Could you do that please in this or. I really appreciate your efforts. I guess it will take less time then we spend writing this comments and protect us with potential binding/classloading/configuration issues.

Reply:
I extracted the docker auth-related classes to a new `docker-auth` maven module.
@skabashnyuk does it correspond to what you were expecting ?